### PR TITLE
Pre-commit hooks and some other tidying

### PR DESCRIPTION
**Pull request by gausie**
- Remove unused pre-commit config (lets use husky), add interactive tools for yarn, remove unused babel transforms
- Remove cache stuff for removed babel transforms
- Add husky, lint-staged
- Remove user-specific vscode settings

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
Comment (Rinn) [inline .husky/pre-commit:1]: is this usable on windows?

File: packages/garbo-lib/package.json [hunk 2]
```diff
@@ -7,7 +7,14 @@
   "scripts": {
     "build": "echo \"Nothing to do\"",
     "format": "eslint src --fix && prettier --write .",
-    "lint": "eslint src && prettier --check ."
+    "lint": "eslint src && prettier --check .",
+    "pre-commit": "lint-staged"
```
Comment (Rinn) [inline packages/garbo-lib/package.json:11]: Should probably still have madge to check for circular dependencies (also in CI?)

Comment (gausie) [reply packages/garbo-lib/package.json:11]: Yeah sure. I'll do it later though.

File: .husky/pre-commit (hunk 1 above)
Comment (gausie) [reply .husky/pre-commit:1]: Yes

**Pull request merged**